lib/widgets/twake_components/twake_smart_refresher.dart

```diff
@@ -1,59 +1,131 @@
+  ScrollController get scrollController => widget.controller.scrollController;
+
+  @override
+  void initState() {
+    super.initState();
+    scrollController.addLoadMoreListener(onLoading);
+  }
+
+  @override
+  void dispose() {
+    scrollController.dispose();
+    super.dispose();
+  }
+
+  Future<void> onRefresh() async {
+    if (widget.controller.isRefeshing) return;
+    widget.controller.onRefresh();
+    widget.onRefresh?.call();
+  }
+
+  Future<void> onLoading() async {
+    if (widget.controller.isLoading) return;
+    widget.controller.onLoading();
+    await widget.onLoading!();
+  }
+
   @override
   Widget build(BuildContext context) {
-    return SmartRefresher(
-      enablePullUp: enablePullUp,
-      enablePullDown: enablePullDown,
-      header: CustomHeader(
-        builder: (context, mode) {
-          if (mode != RefreshStatus.refreshing) {
-            return const SizedBox.shrink();
-          }
-          return const Padding(
-            padding: EdgeInsets.all(16.0),
-            child: Center(
-              child: CircularProgressIndicator(),
+    return _TwakeSmartRefresherView(
+      controller: this,
+      refreshController: widget.controller,
+      slivers: widget.slivers,
+    );
+  }
+}
+
+class _TwakeSmartRefresherView extends StatelessWidget {
+  const _TwakeSmartRefresherView({
+    required this.controller,
+    required this.refreshController,
+    required this.slivers,
+  });
+  final List<Widget> slivers;
+  final TwakeRefreshController refreshController;
+  final _TwakeSmartRefresherController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return RefreshIndicator(
+      onRefresh: controller.onRefresh,
+      child: CustomScrollView(
+        controller: controller.scrollController,
+        slivers: [
+          ValueListenableBuilder(
+            valueListenable: refreshController.refreshNotifier,
+            builder: (context, refreshing, child) => SliverToBoxAdapter(
+              child: refreshing ? const _LoadingIndicator() : const SizedBox(),
             ),
-          );
-        },
-      ),
-      footer: CustomFooter(
-        builder: (context, mode) {
-          if (mode != LoadStatus.loading) {
-            return const SizedBox.shrink();
-          }
-          return const Padding(
-            padding: EdgeInsets.all(16.0),
-            child: Center(
-              child: CircularProgressIndicator(),
+          ),
+          ...slivers,
+          ValueListenableBuilder(
+            valueListenable: refreshController.loadNotifier,
+            builder: (context, loading, child) => SliverToBoxAdapter(
+              child: loading ? const _LoadingIndicator() : const SizedBox(),
             ),
-          );
-        },
+          )
+        ],
       ),
-      controller: controller,
-      onRefresh: onRefresh,
-      onLoading: onLoading,
-      child: child,
+    );
+  }
+}
+
+class _LoadingIndicator extends StatelessWidget {
```

Comment:
this appear in alot of places, can you twake-ize it for all loading indicator ?

Reply:
done